src/app/shared/component/modal/modal.component.ts

```diff
@@ -0,0 +1,43 @@
+import { Component, OnDestroy, OnInit, Inject } from '@angular/core';
+import { carDetailsService } from '../../../services/car-details.service';
+import { Subscription } from 'rxjs';
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
+
+@Component({
+  selector: 'app-modal',
+  templateUrl: './modal.component.html',
+  styleUrl: './modal.component.css',
+})
+export class ModalComponent implements OnInit, OnDestroy {
+  carNames: string[] = [];
+  carSubscription: Subscription | undefined;
+  dialogRef: any;
+
+  constructor(
+    private carService: carDetailsService,
+    public _dialogRef: MatDialogRef<ModalComponent>,
+    @Inject(MAT_DIALOG_DATA) public data: any
+  ) {}
+  ngOnInit(): void {
+    this.getCarNames();
+  }
+  getCarNames() {
+    this.carSubscription = this.carService.getCarList().subscribe({
```

Comment:
Subskrypcje, które są na HttpClient, nie trzeba odsubskrybować. Angular sam je usunie. 
https://netbasal.com/when-to-unsubscribe-in-angular-d61c6b21bad3
Podrzucam artykuł, który opisuje, kiedy trzeba robić unsubscribe a kiedy nie